packages/logic-utils/src/utils/documentation.ts

```diff
@@ -0,0 +1,191 @@
+      matcher: /<b>(.+?)<\/b>/g,
+      replacement: { markdown: "**$1**", plaintext: "$1" },
+    },
+    {
+      matcher: /<strong>(.+?)<\/strong>/g,
+      replacement: { markdown: "**$1**", plaintext: "$1" },
+    },
+
+    // Emphasis
+    {
+      matcher: /<em>(.+?)<\/em>/g,
+      replacement: { markdown: "***$1***", plaintext: "$1" },
+    },
+
+    // Headers
+    {
+      matcher: /<h1>(.+?)<\/h1>/g,
+      replacement: { markdown: "\n# $1\n\n", plaintext: "\n$1\n" },
+    },
+    {
+      matcher: /<h2>(.+?)<\/h2>/g,
+      replacement: { markdown: "\n## $1\n\n", plaintext: "\n$1\n" },
+    },
+    {
+      matcher: /<h3>(.+?)<\/h3>/g,
+      replacement: { markdown: "\n### $1\n\n", plaintext: "\n$1\n" },
+    },
+    {
+      matcher: /<h4>(.+?)<\/h4>/g,
+      replacement: { markdown: "\n#### $1\n\n", plaintext: "\n$1\n" },
+    },
+
+    // Lists
+    {
+      matcher: /<li>(.+?)<\/li>/g,
+      replacement: { markdown: "\n* $1", plaintext: "\n* $1" },
+    },
+    { matcher: /<ul>/g, replacement: { markdown: "", plaintext: "" } },
+    { matcher: /<\/ul>/g, replacement: { markdown: "\n\n", plaintext: "\n" } },
+    // TODO should we handle ordered lists (ol)?
+
+    // Code value
+    {
+      matcher: /<code>(.+?)<\/code>/g,
+      replacement: { markdown: "`$1`", plaintext: "$1" },
+    },
+
+    // Code block
+    {
+      matcher: /<pre>([^]+?)<\/pre>/g,
+      replacement: {
+        markdown: "\n```javascript\n$1```\n",
+        plaintext: "\n$1\n",
+      },
+    },
+
+    // Line break
+    { matcher: /<br\/>/g, replacement: { markdown: "\n", plaintext: "\n" } },
+    { matcher: /<br>/g, replacement: { markdown: "\n", plaintext: "\n" } },
+    { matcher: /<\/br>/g, replacement: { markdown: "\n", plaintext: "\n" } },
+
+    // HTML Escaping, e.g. "&lt;View" --> "<View"
+    // Note: this doesn't replace all html-encoded characters, only the most common ones
+    {
+      matcher: /.*/gm,
+      replacement: { markdown: unescape, plaintext: unescape },
+    },
+
+    // Links
+    // Assuming links are of the form: {@link <type>[ <text>]}
+    // Where the type doesn't contain whitespace, and neither the type nor text contain the "}" character
+    {
+      matcher: /{@link (([^\s}]+)\s)?([^}]+)}/g,
+      replacement: {
+        markdown: (all, _, type, text): string => {
+          return `[${text}](${getLink(model, type ?? text)})`;
+        },
+        plaintext: "$3",
+      },
+    },
+  ];
+
+  let contents = jsdoc;
+  forEach(tagMatcherToReplacement, (_) => {
+    const replacement = _.replacement[target];
+    // Workaround for typescript signature definition
+    if (typeof replacement === "string") {
+      contents = contents.replace(_.matcher, replacement);
+    } else {
+      contents = contents.replace(_.matcher, replacement);
+    }
+  });
+
+  return contents;
+}
+
+export function getLink(model: UI5SemanticModel, link: string): string {
+  if (link.startsWith("http:") || link.startsWith("https:")) {
```

Comment:
Should we convert all http links to http**s** links? is this safe?

Reply:
I don't want to convert links if they're absolute... definitely not when we don't know where they link to